ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -180,6 +181,16 @@ interface UtxoTransactionBuilder {
      */
     fun setTimeWindowBetween(from: Instant, until: Instant): UtxoTransactionBuilder
 
+    /**
+     * Sets the type of token claim behavior when finalizing a transaction
+     *
+     * @param behaviorType The [TokenClaimBehaviorType] for the transaction. If unspecified defaults to
+     * [TokenClaimBehaviorType.CLAIM]
+
+     * @return Returns a [UtxoTransactionBuilder] including the specified claim behavior.
+     */
+    fun setTokenClaimBehavior(behaviorType: TokenClaimBehaviorType): UtxoTransactionBuilder
```

Comment:
Out of curiosity, why is not this a finality flow parameter?

Reply:
I agree that it should be part of `finalize`.

Do we also want something similar for `receiveFinality`, is there a benefit in having them receive the tokens as well, as they will also have them stored and available to spend. This would mean receive finality would have both a `TokenClaimBehaviourType` argument and also a return type of `FinalizationResult`.